### ISSUE TITLE
[Feature] Make layout persistent

### ISSUE THREAD

**Issue by robsontenorio** [mtolhuys/laravel-schematics#12]
Thanks for your time! Nice work!

Would be nice to have a persistent layout (diagram).

- Open http://localhost:8080/schematics
- Move around some objects
- Open a icognito browser http://localhost:8080/schematics
- Diagram layout has lost.

**Comment by mtolhuys** [comment]
Hi again @robsontenorio! :)

Thanks! 

The diagram saving relies on the localStorage which gets lost when you open a incognito screen. You could use the export / import to restore it though.

Maybe in the future I could save the diagrams somewhere else so they'll go along with incognito or browser switching. I'll have to think of a implementation.

**Comment by ArtNous** [comment]
I think you could use a database, or Redis to save the layout.

**Comment by mtolhuys** [comment]
@ArtNous Yeah, could definitely be an option. I went for localStorage to make it the least "intrusive" as it could be. 

Thinking about it I'm actually leaning towards using backend caching.

Will be a heavy change since pretty much everything you do gets stored in it (dragging positions etc.)

**Comment by robsontenorio** [comment]
You could keep all current logic using localStorage. But just send back to backend and save all changes on dedicated table.

In fact would be necessary to have a persitent mechanism, like a table.

EDIT: or maybe a simple file on proper location. 

![Captura de tela de 2020-03-06 19 36 07](https://user-images.githubusercontent.com/118955/76127905-bf4d0e80-5fe1-11ea-9e79-0d66cfbc83e8.png)

**Comment by ArtNous** [comment]
That's right, It would be a big change, but I think it is necessary, we're here to support and help.

**Comment by mtolhuys** [comment]
Thanks guys! All contributions will be fully credited of course. Just have to make a nice section for it in the README as I didn't thought this would happen so fast. :)

I'll take the suggestions with me.

**Comment by mtolhuys** [comment]
At the moment I'm working on a "hybrid" solution for this. I'm thinking of a save button in the right upper corner which will cache the localStorage. With that, if the localStorage is empty, it could fill it with it. This way we'll prevent a request bombardment. :)
